snippets/go/metadata.go

```diff
@@ -0,0 +1,73 @@
+package example
+
+import (
+	"encoding/json"
+	"log"
+
+	"github.com/breez/breez-sdk-go/breez_sdk"
+)
+
+func SetPaymentMetadata() {
+	// ANCHOR: set-payment-metadata
+	sdk.SetPaymentMetadata("target-payment-hash", `{"myCustomValue":true}`)
+	// ANCHOR_END: set-payment-metadata
+}
+
+func FilterPaymentMetadata() {
+	// ANCHOR: filter-payment-metadata
+	metadataFilters := []breez_sdk.PaymentMetadata {
+    	{ json_path: "myCustomValue", json_value: "true" },
```

Comment:
Go public struct variables should be pascal case
```suggestion
    	{ JsonPath: "myCustomValue", JsonValue: "true" },
```